hazelcast/src/test/java/com/hazelcast/test/OverridePropertyRule.java

```diff
@@ -0,0 +1,67 @@
+package com.hazelcast.test;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+
+/**
+ * Set or clear a property before running a test. The property will be restored once the test is finished.
+ *
+ */
+public final class OverridePropertyRule implements TestRule {
+    private final String propertyName;
+    private final String value;
+
+    private OverridePropertyRule(String propertyName, String value) {
+        this.propertyName = propertyName;
+        this.value = value;
+    }
+
+    /**
+     * Clears the property.
+     *
+     * @param propertyName system property to clear
+     * @return instance of the rule
+     */
+    public static OverridePropertyRule clear(String propertyName) {
+        return new OverridePropertyRule(propertyName, null);
+    }
+
+
+    /**
+     * Set the property to a newValue
+     *
+     * @param propertyName system property to set
+     * @param newValue value to set
+     * @return instance of the rule
+     */
+    public static OverridePropertyRule set(String propertyName, String newValue) {
```

Comment:
Could we have a `set(HazelcastProperty property, String newValue)` method? Then you could use it like this in the test (with a static import):
```java
@Rule
public final OverridePropertyRule overridePropertyRule = set(MAX_JOIN_SECONDS, "10");
```

Reply:
I don't really like introducing a dependency on Hazelcast in this rule. you can always use 
`public final OverridePropertyRule overridePropertyRule = set(MAX_JOIN_SECONDS.getName(), "10");
 Write`